Lib/tempfile.py

```diff
@@ -822,8 +822,15 @@ def resetperms(path):
 
                     try:
                         _os.unlink(path)
+                    except IsADirectoryError:
+                        cls._rmtree(path, ignore_errors=ignore_errors)
                     # PermissionError is raised on FreeBSD for directories
-                    except (IsADirectoryError, PermissionError):
+                    except PermissionError:
+                        # On Windows, _rmtree will cause IsADirectoryError
```

Comment:
I don't understand the comment. There is now an explicit `except IsADirectoryError:`. We are handling PermissionError. I don't understand the relationship between PermissionError and IsADirectoryError.

Reply:
My comment needs improving then :). The current bug is that calling `_rmtree` when there's a `PermissionError` raises `IsADirectoryError`, which masks the original ``PermissionError`. I was trying to convey that but seems like the comment wasn't clear.